tests/test_corpus.py

```diff
@@ -114,5 +111,31 @@ def test_interesting_results_are_added_to_database(args):
         }
 
 
+def test_corpus_resets_branch_counts_on_new_coverage():
+    hypothesis_db = InMemoryExampleDatabase()
+
+    @given(st.integers())
+    @settings(database=hypothesis_db)
+    def test_a(x):
+        if x == 2:
+            pass
+
+    process = FuzzProcess.from_hypothesis_test(test_a)
+    process._start_phase(Phase.GENERATE)
+    for count in range(1, 10):
+        process._run_test_on(
+            ConjectureData.for_choices([1]), collector=Collector(test_a)
+        )
+        # we keep incrementing arc counts if we don't find new coverage
+        assert process.corpus.branch_counts == {((3, 11), (3, 11)): count}
+
+    process._run_test_on(ConjectureData.for_choices([2]), collector=Collector(test_a))
+    # our arc counts should get reset whenever we discover a new branch
+    assert process.corpus.branch_counts == {
```

Comment:
FYI: we do reset branch counts (soon behavior counts) every time we find a new branch (...behavior), and I think that's the right call at the moment.  Eventually we'll want to use a fancier estimator which can leverage information from the slightly-different prior distribution; iirc Boehme et al studied that at some point.

Reply:
sounds good! This test isn't really related to this PR, I was just trying to understand the corpus logic (because I thought it had a bug, and then discovered it didn't).